source/renderer/app/components/wallet/transactions/FilterDialog.js

```diff
@@ -0,0 +1,535 @@
+} from '../../../stores/TransactionsStore';
+import {
+  DateRangeTypes,
+  emptyTransactionFilterOptions,
+} from '../../../stores/TransactionsStore';
+import TinyCheckbox from '../../widgets/forms/TinyCheckbox';
+import TinySelect from '../../widgets/forms/TinySelect';
+import TinyInput from '../../widgets/forms/TinyInput';
+import TinyDatePicker from '../../widgets/forms/TinyDatePicker';
+import TinyButton from '../../widgets/forms/TinyButton';
+import DialogCloseButton from '../../widgets/DialogCloseButton';
+import Dialog from '../../widgets/Dialog';
+import globalMessages from '../../../i18n/global-messages';
+import styles from './FilterDialog.scss';
+
+const messages = defineMessages({
+  filterBy: {
+    id: 'wallet.transaction.filter.filterBy',
+    defaultMessage: '!!!Filter by',
+    description: 'Title of filter panel.',
+  },
+  reset: {
+    id: 'wallet.transaction.filter.reset',
+    defaultMessage: '!!!reset',
+    description: 'Reset link text on filter panel.',
+  },
+  type: {
+    id: 'wallet.transaction.filter.type',
+    defaultMessage: '!!!Type',
+    description: 'Filter Type.',
+  },
+  incoming: {
+    id: 'wallet.transaction.filter.incoming',
+    defaultMessage: '!!!Incoming',
+    description: 'Incoming filter type.',
+  },
+  outgoing: {
+    id: 'wallet.transaction.filter.outgoing',
+    defaultMessage: '!!!Outgoing',
+    description: 'Outgoing filter type.',
+  },
+  dateRange: {
+    id: 'wallet.transaction.filter.dateRange',
+    defaultMessage: '!!!Date range',
+    description: 'Date range of filter.',
+  },
+  thisWeek: {
+    id: 'wallet.transaction.filter.thisWeek',
+    defaultMessage: '!!!This week',
+    description: 'This week date range of filter.',
+  },
+  thisMonth: {
+    id: 'wallet.transaction.filter.thisMonth',
+    defaultMessage: '!!!This month',
+    description: 'This month date range of filter.',
+  },
+  thisYear: {
+    id: 'wallet.transaction.filter.thisYear',
+    defaultMessage: '!!!This year',
+    description: 'This year date range of filter.',
+  },
+  customDateRange: {
+    id: 'wallet.transaction.filter.customDateRange',
+    defaultMessage: '!!!Custom date range',
+    description: 'Custom date range of filter.',
+  },
+  amountRange: {
+    id: 'wallet.transaction.filter.amountRange',
+    defaultMessage: '!!!Amount range',
+    description: 'Amount range of filter.',
+  },
+  amountUnit: {
+    id: 'global.unit.ada',
+    defaultMessage: '!!!ADA',
+    description: 'Amount unit.',
+  },
+  filter: {
+    id: 'wallet.transaction.filter.filter',
+    defaultMessage: '!!!Filter',
+    description: 'Filter button label.',
+  },
+});
+
+const calculateDateRange = (
+  dateRange: string,
+  customDateRange: { customFromDate: string, customToDate: string }
+) => {
+  const { customFromDate, customToDate } = customDateRange;
+  let fromDate = null;
+  let toDate = null;
+
+  if (dateRange === DateRangeTypes.ALL) {
+    fromDate = '';
+    toDate = '';
```

Comment:
@tomothespian No, this is needed because those two variables need to be cleared again after user applied filter with some `fromDate` and `toDate` values.
You can test it on app.

Reply:
@yakovkaravelov ok cool, then ignore this comment 👍